src/server/project.ts

```diff
@@ -310,6 +311,13 @@ namespace ts.server {
             this.lsHost.dispose();
             this.lsHost = undefined;
 
+            // Clean up file watchers waiting for missing files
+            for (const p of this.missingFilesMap.getKeys()) {
+                this.missingFilesMap.get(p).close();
+                this.missingFilesMap.remove(p);

```

Comment:
What's the point of removing items one-by-one if you just set `this.missingFileMap = undefined;` at the end?